### PR TITLE
Add `metalctl console` subcommand

### PR DESCRIPTION
**Pull request by afritzler**
# Proposed Changes

- Add a `metalctl console` subcommand with which you are able to connect to a `Servers` serial console
- Factor out `bmcutils` into an own package

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -9,34 +9,34 @@ flowchart LR
     subgraph Out-of-Band Network
         EndpointReconciler
     end
-    EndpointReconciler -- Discovers --> EndpointCRD
+    EndpointReconciler -- Discovers --> Endpoint
     EndpointCRD -- Uses --> MACPrefixDatabase
```
Comment (SzymonSAP) [inline docs/architecture.md:13]:
```suggestion
    Endpoint -- Uses --> MACPrefixDatabase
```
?

File: docs/architecture.md [hunk 2]
```diff
@@ -9,34 +9,34 @@ flowchart LR
...
 
-    ServerReconciler -- Manages --> ServerCRD
+    ServerReconciler -- Manages --> Server
     ServerReconciler -- Uses --> metalprobe
     ServerReconciler -- Waits for --> ServerBootConfiguration
 
-    ServerClaimReconciler -- Manages --> ServerClaimCRD
-    ServerClaimCRD -- References --> ServerCRD
+    ServerClaimReconciler -- Manages --> ServerClaim
+    ServerClaimCRD -- References --> Server
```
Comment (SzymonSAP) [inline docs/architecture.md:25]:
```suggestion
    ServerClaim -- References --> Server
```

**Review by SzymonSAP — COMMENTED** [review]
The code is well written and there is not much to point out but I found 3 fast fixes to be done. Good job! ;)

File: cmd/metalctl/app/console.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+	consoleCmd := &cobra.Command{
+		Use:   "console",
+		Short: "Access the serial console of a Server",
+		RunE:  runConsole,
+	}
+
+	consoleCmd.Flags().StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig.")
+	consoleCmd.Flags().IntVar(&serialConsoleNumber, "serial-console-number", 1, "Serial console number.")
+
+	if verbose {
```
[the remark this answers is not in the document]
Comment (afritzler) [reply cmd/metalctl/app/console.go]: Thx for pointing this out!

File: internal/console/console.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		}
+
+		return &Config{
+			BMCAddress: address,
+			Username:   username,
+			Password:   password,
+		}, nil
+	}
+
+	return nil, nil
```
Comment (Nuckal777) [inline internal/console/console.go:62]: Shouldn't this return an error?

File: internal/console/suite_test.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+)
+
+func TestControllers(t *testing.T) {
+	SetDefaultConsistentlyPollingInterval(pollingInterval)
+	SetDefaultEventuallyPollingInterval(pollingInterval)
+	SetDefaultEventuallyTimeout(eventuallyTimeout)
+	SetDefaultConsistentlyDuration(consistentlyDuration)
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "Controller Suite")
```
Comment (Nuckal777) [inline internal/console/suite_test.go:47]: Copy Pasta: `"Console Suite"`

File: internal/console/console.go (hunk 4 above)
Comment (afritzler) [reply internal/console/console.go:62]: Good point.

**Review by Nuckal777 — APPROVED** [review]

**Pull request merged**